android/src/main/java/com/tadasr/IOTWifi/IOTWifiModule.java

```diff
@@ -88,6 +92,9 @@ private void connectToWifi(String ssid, String passphrase, Boolean isWEP, Callba
             }
             try {
                 Thread.sleep(3000);
+                if (bindNetwork) {
```

Comment:
Any specific reason you are binding after the sleep?

Reply:
I assumed the sleep was added to wait for the network to be connected and available, originally this was not checking that the bound network has the correct SSID, but now that it is that bind could happen elsewhere.